packages/openai/src/responses/openai-responses-language-model.ts

```diff
@@ -459,10 +465,23 @@ export class OpenAIResponsesLanguageModel implements LanguageModelV2 {
                 text: value.delta,
               });
             } else if (isResponseReasoningSummaryTextDeltaChunk(value)) {
+              // When the summary_index changes, mark the end of the previous
+              // reasoning part before streaming the new one.
+              if (
+                lastReasoningSummaryIndex !== null &&
+                value.summary_index !== lastReasoningSummaryIndex
+              ) {
+                controller.enqueue({ type: 'reasoning-part-finish' });
+              }
+
+              lastReasoningSummaryIndex = value.summary_index;
```

Comment:
this assumes that reasoning chunks arrive sequentially (necessary because of how the language model is chunks are structured). fine for now but something to potentially look at in the future (for something like language model v3)